launch/launch/actions/execute_process.py

```diff
@@ -426,40 +426,58 @@ def __on_process_stdin(
-                self.__stdout_logger.info(
-                    self.__output_format.format(line=line[:-len(os.linesep)], this=self)
-                )
-            else:
-                last_line = line
-                break
-        self.__stdout_buffer.seek(0)
-        self.__stdout_buffer.truncate(0)
-        if last_line is not None:
-            self.__stdout_buffer.write(last_line)
+        to_write = event.text.decode(errors='replace')
+        try:
+            self.__stdout_buffer.write(to_write)
+            self.__stdout_buffer.seek(0)
+            last_line = None
+            for line in self.__stdout_buffer:
+                if line.endswith(os.linesep):
+                    self.__stdout_logger.info(
+                        self.__output_format.format(line=line[:-len(os.linesep)], this=self)
+                    )
+                else:
+                    last_line = line
+                    break
+            self.__stdout_buffer.seek(0)
+            self.__stdout_buffer.truncate(0)
+            if last_line is not None:
+                self.__stdout_buffer.write(last_line)
+        except ValueError:
+            # __stdout_buffer was probably closed by __flush_buffers on shutdown.  Output without
+            # buffering.
+            self.__stdout_logger.info(
+                self.__output_format.format(line=to_write, this=self)
+            )
+
 
     def __on_process_stderr(
         self, event: ProcessIO
     ) -> Optional[SomeActionsType]:
-        self.__stderr_buffer.write(event.text.decode(errors='replace'))
-        self.__stderr_buffer.seek(0)
-        last_line = None
-        for line in self.__stderr_buffer:
-            if line.endswith(os.linesep):
-                self.__stderr_logger.info(
-                    self.__output_format.format(line=line[:-len(os.linesep)], this=self)
-                )
-            else:
-                last_line = line
-                break
-        self.__stderr_buffer.seek(0)
-        self.__stderr_buffer.truncate(0)
-        if last_line is not None:
-            self.__stderr_buffer.write(last_line)
+        to_write = event.text.decode(errors='replace')
+        try:
+            self.__stderr_buffer.write(to_write)
+            self.__stderr_buffer.seek(0)
+            last_line = None
+            for line in self.__stderr_buffer:
+                if line.endswith(os.linesep):
+                    self.__stderr_logger.info(
+                        self.__output_format.format(line=line[:-len(os.linesep)], this=self)
+                    )
+                else:
+                    last_line = line
+                    break
+            self.__stderr_buffer.seek(0)
+            self.__stderr_buffer.truncate(0)
+            if last_line is not None:
+                self.__stderr_buffer.write(last_line)
+        except ValueError:
+            # __stderr buffer was probably closed by __flush_buffers on shutdown.  Output without
+            # buffering.
```

Comment:
@pbaughman same as above.

Reply:
@hidmic Yeah, I was coming at this from the approach of "this is a race condition" but this method is synchronized with the one that closes the buffers, right?  They're not going to get closed out from under us while the method is running?

Reply:
It's `asyncio`, so no, that won't happen. There're no guarantees regarding execution order, and that's what probably caused this issue in the first place.